packages/devtools_app/test/extensions/extension_settings_test.dart

```diff
@@ -0,0 +1,215 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/extensions/extension_settings.dart';
+import 'package:devtools_shared/devtools_extensions.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../test_infra/matchers/matchers.dart';
+import '../test_infra/test_data/extensions.dart';
+
+void main() {
+  late ExtensionSettingsDialog dialog;
+
+  final stubEnabledStates = <String, ValueNotifier<ExtensionEnabledState>>{};
+
+  Future<void> setUpExtensionService(
+    List<DevToolsExtensionConfig> extensions,
+    // ignore: avoid-redundant-async, false positive
+  ) async {
+    final mockExtensionService = MockExtensionService();
+    setGlobal(ExtensionService, mockExtensionService);
+    when(mockExtensionService.availableExtensions)
+        .thenReturn(ImmediateValueNotifier(extensions));
+
+    stubEnabledStates.clear();
+    for (final e in extensions) {
+      stubEnabledStates[e.name.toLowerCase()] =
+          ValueNotifier<ExtensionEnabledState>(ExtensionEnabledState.none);
+      when(mockExtensionService.enabledStateListenable(e.name))
+          .thenReturn(stubEnabledStates[e.name.toLowerCase()]!);
+      when(mockExtensionService.enabledStateListenable(e.name.toLowerCase()))
+          .thenReturn(stubEnabledStates[e.name.toLowerCase()]!);
+      when(mockExtensionService.setExtensionEnabledState(e, enable: true))
+          .thenAnswer((_) async {
+        stubEnabledStates[e.name.toLowerCase()]!.value =
+            ExtensionEnabledState.enabled;
+        return Future.value();
```

Comment:
Since this function is marked `async`, I think you can just drop the `return Future.value()` here and below.